javascript/URDFLoader.js

```diff
@@ -515,35 +546,27 @@ class URDFLoader {
                 rpy = this._processTuple(n.getAttribute('rpy'));
 
             } else if (type === 'material') {
+                const materialName = n.getAttribute('name');
+                if (n.children.length) {
+                    this.forEach(n.children, c => {
+                        switch (c.nodeName.toLowerCase()) {
+
+                            case 'color':
+                                this._processMaterial(material, 'color', c.getAttribute('rgba'));
+                                break;
+                            case 'texture':
+                                if (materialName in materialMap) {
+                                    material.copy(materialMap[materialName]);
+                                } else {
+                                    this._processMaterial(material, 'texture', c.getAttribute('filename'), packages, path);
+                                }
 
-                this.forEach(n.children, c => {
-
-                    if (c.nodeName.toLowerCase() === 'color') {
-
-                        const rgba = c.getAttribute('rgba')
-                            .split(/\s/g)
-                            .map(v => parseFloat(v));
-
-                        material.color.r = rgba[0];
-                        material.color.g = rgba[1];
-                        material.color.b = rgba[2];
-                        material.opacity = rgba[3];
-
-                        if (material.opacity < 1) material.transparent = true;
-
-                    } else if (c.nodeName.toLowerCase() === 'texture') {
-
-                        const filename = c.getAttribute('filename').replace(/^(package:\/\/)/, '');
-                        const filePath = this._resolvePackagePath(packages, filename, path);
-
-                        material.map = this._textureloader.load(filePath);
-
-                    }
-
-                });
-
+                        }
+                    });
+                } else {
+                    material.copy(materialMap[materialName]);
```

Comment:
Using `Material.copy` concerns me a little bit because it does a lot more than just copy the color and texture and it doesn't work well when using different material types. `new THREE.MeshStandardMaterial().copy(new THREE.MeshLambertMaterial())` throws an error, for example, so it's not always safe to call copy on a material of a different type. And if someone has created a custom material as a part of the mesh load function then things like the specular values will be overwritten. It would overwrite the color even if it wasn't specified in the URDF, too.

How do you feel about copying just those specific values:

```js
const sharedMatData = materialMap[materialName];
if (sharedMatData.color) material.color.copy(sharedMatData.color);
if (sharedMatData.map) material.map = sharedMatData.map;
```

And then the material map could store just those preprocessed values in an object. The color and texture node parsing could be pulled out into separate functions that return the texture and color objects.